spec/01-unit/27-queue_spec.lua

```diff
@@ -392,6 +416,128 @@ describe("plugin queue", function()
     )
   end)
 
+  it("works when time is moved forward while items are being queued", function()
+    local number_of_entries = 1000
+    local number_of_entries_processed = 0
+    now_offset = 1
+    for i = 1, number_of_entries do
+      Queue.enqueue(
+        queue_conf({
+          name = "time-forwards-adjust",
+          max_batch_size = 10,
+          max_coalescing_delay = 10,
+        }),
+        function(_, entries)
+          number_of_entries_processed = number_of_entries_processed + #entries
+          return true
+        end,
+        nil,
+        i
+      )
+      now_offset = now_offset + now_offset
```

Comment:
What's the logic behind this?

Reply:
We want to test whether the queue still works if entries are added to the queue while the current system time is changed (forward in this case).  This is just a simple way to achieve that, but I'm open to suggestions what would be better as a test.